bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java

```diff
@@ -233,11 +233,16 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         TeslaChannelSelector selector = TeslaChannelSelector.getValueSelectorFromChannelID(channelID);
 
         if (command instanceof RefreshType) {
-            if (isAwake()) {
-                // Request the state of all known variables. This is sub-optimal, but the requests get scheduled and
-                // throttled so we are safe not to break the Tesla SLA
-                requestAllData();
+            if (!isAwake()) {
+                logger.debug("Waking vehicle to refresh all data");
```

Comment:
> to refresh all data

Note that REFRESH is sent per channel - it is therefore also only expected to receive a state update for the requested channel.
Furthermore, the user might write a rule where he sends a REFRESH to 20 channels at the same time. Do you know how `wakeUp()` behaves, if it gets 20 calls at the same moment? Do we check whether a wake up has just been sent and we can ignore any further ones for a certain time frame?

Reply:
We do not currently check the success of a wakeup request. The API also does not really provide this, you send the wakeup command and then need to wait/poll the vehicle state until it comes online.

If 20 channels are refreshed at the same time and the car is asleep/offline, it will receive 20 wake up commands

Reply:
> it will receive 20 wake up commands

That's what I mean - we know that waking up takes a while, so sending 20 commands within a few milliseconds sounds rather problematic. And if the throttler queues them for a long period, it doesn't make the situation much better. Imho we should just ignore additional wakeup commands that are received within a certain timeframe (maybe somewhere around 1-10 seconds).

Reply:
I agree, I can add a 10s period where we ignore all wakeUp commands after we have received one already. However I'd first like to clear up the confusion with the other comments

Reply:
Ok, this can also be done in a follow-up PR.